### ISSUE TITLE
`ObanWeb` crashes the test suite

### ISSUE THREAD

**Issue by lucasmazza** [sorentwo/oban#109]
### Environment

- oban `0.11.1` / oban_web `0.6.0`
- postgresql 11.1
- elixir `1.9.4` & Erlang/OTP 22

I'm trying `oban_web` with @philss on our app and we faced a setup issue when running `mix test` after installing and trying out the UI on the browser successfully, where all our repo related tests are failing:

```
bash-5.0# mix test
18:22:25.695 [error] Postgrex.Protocol (#PID<0.425.0>) disconnected: ** (DBConnection.ConnectionError) #PID<0.92.0> checked in the connection owned by #PID<0.453.0>

#PID<0.92.0> triggered the checkin at location:

    (kernel) code_server.erl:139: :code_server.call/1
    (elixir) src/elixir_compiler.erl:83: :elixir_compiler.dispatch/4
    (elixir) src/elixir_compiler.erl:65: :elixir_compiler.compile/3
    (elixir) src/elixir_compiler.erl:25: anonymous fn/4 in :elixir_compiler.quoted/3
    (elixir) src/elixir_lexical.erl:17: :elixir_lexical.run/2
    (elixir) src/elixir_compiler.erl:23: :elixir_compiler.quoted/3
    (elixir) lib/code.ex:813: Code.require_file/2
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2


Client #PID<0.453.0> is still using a connection from owner at location:

    (kernel) code_server.erl:139: :code_server.call/1
    (kernel) error_handler.erl:41: :error_handler.undefined_function/3
    (ecto) lib/ecto/repo/queryable.ex:14: Ecto.Repo.Queryable.all/3
    (oban_web) lib/oban_web/stats.ex:202: ObanWeb.Stats.fetch_node_counts/1
    (oban_web) lib/oban_web/stats.ex:128: anonymous fn/1 in ObanWeb.Stats.handle_info/2
    (ecto_sql) lib/ecto/adapters/sql.ex:898: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection) lib/db_connection.ex:717: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2

The connection itself was checked out by #PID<0.453.0> at location:

    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7


18:22:25.757 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.ConnectionError) connection is closed because of an error, disconnect or timeout
    (ecto_sql) lib/ecto/adapters/sql.ex:629: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql) lib/ecto/adapters/sql.ex:562: Ecto.Adapters.SQL.execute/5
    (ecto) lib/ecto/repo/queryable.ex:177: Ecto.Repo.Queryable.execute/4
    (ecto) lib/ecto/repo/queryable.ex:17: Ecto.Repo.Queryable.all/3
    (oban_web) lib/oban_web/stats.ex:202: ObanWeb.Stats.fetch_node_counts/1
    (oban_web) lib/oban_web/stats.ex:128: anonymous fn/1 in ObanWeb.Stats.handle_info/2
    (ecto_sql) lib/ecto/adapters/sql.ex:898: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection) lib/db_connection.ex:717: DBConnection.run/3
Last message: {:continue, :start}
18:22:25.791 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.464.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.802 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.465.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.812 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.466.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.818 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.468.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.822 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.469.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.826 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.470.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.832 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.471.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.844 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.473.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.848 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.474.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.853 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.475.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.859 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.476.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.863 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.478.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.864 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.479.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.866 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.480.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
18:22:25.867 [error] GenServer ObanWeb.Stats terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.481.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

    (db_connection) lib/db_connection.ex:746: DBConnection.run/3
    (oban_web) lib/oban_web/stats.ex:127: ObanWeb.Stats.handle_info/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
```

After this, all conn cases tests fail with:

```
** (RuntimeError) could not lookup Ecto repo MyApp.Repo because it was not started or it does not exist
stacktrace:
  (ecto) lib/ecto/repo/registry.ex:19: Ecto.Repo.Registry.lookup/1
  (ecto) lib/ecto/adapter.ex:127: Ecto.Adapter.lookup_meta/1
  (ecto_sql) lib/ecto/adapters/sql/sandbox.ex:487: Ecto.Adapters.SQL.Sandbox.lookup_meta!/1
  (ecto_sql) lib/ecto/adapters/sql/sandbox.ex:415: Ecto.Adapters.SQL.Sandbox.checkout/2
  (my_app) test/support/data_case.ex:30: MyApp.DataCase.__ex_unit_setup_0/1
  (my_app) test/support/data_case.ex:1: MyApp.DataCase.__ex_unit__/2
```

**Comment by sorentwo** [comment]
This is something I'm working on.

As a quick fix I suggest that you don't start `ObanWeb` when you are running in the test environment.

**Comment by sorentwo** [comment]
@lucasmazza Fixed with v0.6.1. If you're setting `queues: false` for Oban during test mode then the stats process won't start at all now, preventing this error entirely.

For others who are using `shared` mode exclusively; I've fixed a race condition between stats and the Ecto repo during shutdown.

/cc @sztosz
